### ISSUE TITLE
[Bug] Auto-rotate screen changes current reviewing card

### ISSUE THREAD

**Issue by Feripe-a** [ankidroid/Anki-Android#9360]
###### Reproduction Steps

1.  Turn on "Auto-rotate";
2.  Start reviewing cards;
3.  Rotate the phone;
4. Screen is rotated.

###### Expected Result

Anki would rotate the screen displaying the card in landscape or portrait mode.

###### Actual Result

As shown [in this video](https://streamable.com/kschia), Anki does display the card in landscape or portrait mode but changes the current reviewing card to a different one.

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".
AnkiDroid Version = 2.16alpha7

Android Version = 11 (with [LineageOS](https://lineageos.org/) instead of [MIUI](https://en.miui.com/))

Manufacturer = Xiaomi

Model = Redmi Note 5

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 11; Redmi Note 5 Build/RQ3A.210705.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/91.0.4472.164 Mobile Safari/537.36

ACRA UUID = fc79c16d-49ff-4214-a96c-049159ef42a8

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by prateek-singh-3212** [comment]
Hello @Feripe-a ,
Thanks for reporting the issue but I am unable to reproduce this on my device (Galaxy M51) and the vedio streaming link is not working.

**Comment by Feripe-a** [comment]
>the video streaming link is not working.

Strange. It works just fine here. I've managed to reduce the quality to send the video here.

https://user-images.githubusercontent.com/84395172/127936443-4d406aa1-568d-4e71-ae0e-629034731019.mp4

**Comment by Feripe-a** [comment]
>I am unable to reproduce this on my device (Galaxy M51).

That's why I wrote "Android Version = 11 (with LineageOS instead of MIUI)" because its related to  LineageOS, but I don't know anyone that uses it and uses Anki -- I'm really sorry, I should've mentioned it.

**Comment by prateek-singh-3212** [comment]
>  I should've mentioned it.

No not an issue.

>  I don't know anyone that uses it and uses Anki

Yes I also strongly believes on your thoughts. But i dont have stats so I can't confirm it. 

 This issue is related to OS itself. As far as I know ( please confirm that other app are working fine ?). If that's not the case then this issue will be on ```
Priority -Low 
``` list.

**Comment by Feripe-a** [comment]
>please confirm that other app are working fine?

Only Anki presents that kind of error. I've tasted with all apps I use every day, all of them rotate like they were supposed to.

**Comment by mikehardy** [comment]
I wonder - when you rotate the screen it usually restarts the activity as it represents a configuration change - perhaps there is some part of the bundle that is not saved restored correctly across activity restarts?

@prateek-singh-3212 if you try to reproduce after turning on developer mode in your system settings then toggling "keep activities" to off in the developer settings, you might see this.

It's a classic problem in Android apps (though most handle it now) and for us historically we have not always done a good job with it. Turning off keep activities is how to troubleshoot it then the fix is usually in saveInstanceBundle / onCreate to save/restore the proper variables across screen restart

I used to use LineageOS a lot actually :-), the thing about LineageOS is that it is very very close to stock Android, so this should not be LineageOS specific really, it should be reproducible if we just figure out how.

**Comment by david-allison-1** [comment]
Should be able to repro by toggling night mode

**Comment by mikehardy** [comment]
oh yes - that's another good repro style - for completeness, the reason why night mode would do it as well is because it *also* restarts the Activity - they both will perhaps trigger the same underlying (bad) code pathway

**Comment by Feripe-a** [comment]
I'm glad to read that you three are working looking at it. Keep up with the great work.

**Comment by prateek-singh-3212** [comment]
>  if you try to reproduce after turning on developer mode in your system settings then toggling "keep activities" to off in the developer settings, you might see this.

>Should be able to repro by toggling night mode

Yes, It is reproducable from both methods. Thanks for describing I am not aware about this. I will try to get back on this in future. (Taken Lot's of ISSUES in my Todo 😅 )

Feel free take this if anyone wants to.
